plugins/outputs/kafka/kafka.go

```diff
@@ -186,6 +197,11 @@ func (k *Kafka) Connect() error {
 	}
 	config := sarama.NewConfig()
 
+	kafkaVersion, err := sarama.ParseKafkaVersion(k.Version)
+	if err != nil {
+		return err
+	}
+	config.Version = kafkaVersion
```

Comment:
Is this required for this patch to work, what happens if it is removed?  The documentation seems to indicate that this is used to control some features, what is improved by adding it?

Reply:
If not set this parameter sarama lib return error if we switch to LZ4 compression. 
https://github.com/Shopify/sarama/blob/70f6a705d4a17af059acbc6946fb2bd30762acd7/config_test.go#L199
https://github.com/Shopify/sarama/blob/44e7121d3b5189096ae4ef90c442f5f806c10fc9/config.go#L416

In some places of sarama the 'Version' parameter used to switch to "new" Kafka API that supposed to be more perfomance than "old" one.